lib/cocoapods-core/specification/linter.rb

```diff
@@ -394,6 +394,21 @@ def _validate_test_type(t)
         end
       end
 
+      def _validate_app_host_name(n)
+        unless consumer.requires_app_host?
+          results.add_error('app_host_name', '`requires_app_host` must be set to ' \
+            '`true` when `app_host_name` is specified.')
+        end
+
+        parent_spec_name = (consumer.spec.parent && consumer.spec.parent.name) || consumer.name
+        podspec_name, appspec_name = n.split('/')
+
+        unless podspec_name == parent_spec_name || consumer.dependencies.map(&:name).include?(appspec_name)
+          results.add_error('app_host_name', "Test spec must depend on `#{podspec_name}` " \
```

Comment:
message can include `Test spec `#{consumer.spec.name}` ...` so the user knows explicitly which test spec has the problem.

Alternatively the message can be "The app host name specified by `#{consumer.spec.name}` could not be found. Either explicitly declare it within `#{podspec_name}` or add a dependency to the podspec that specifies it."